students/idcrickmore/lesson03/mailroom.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python3
+
+donors = [
+        ["Galileo Galilei", 348, 8377, 123],
+        ["Giovanni Cassini", 209],
+        ["Christiaan Huygens", 9135, 39],
+        ["Edmond Halley", 399, 1100, 357],
+        ["Edwin Hubble", 1899]
+        ]
+
+
+def menu():
+    print("\n------------------ MENU ------------------\n")
+    print("PLEASE CHOOSE FROM THE FOLLOWING THREE OPTIONS\n")
+    print("1. Send Thank You\n2. Create a Report\n3. Quit\n")
+
+    # the 'option' function calls an appropriate function based on user input
+    def options():
+        # save user input to 'choice' as a lowercase string
+        choice = input("-> ").lower()
+        # user input can be a number or a string
+        # if its a string, just look at the first letter
+        if choice[0] == '1' or choice[0] == 's':
+            thankyou()
+        elif choice[0] == '2' or choice[0] == 'c':
+            report()
+        elif choice[0] == '3' or choice[0] == 'q':
+            print("quiting")
+            quit()
+        else:
+            # if the user input does not match one of the 3 options,
+            # call the menu function again
+            menu()
+
+    options()
+
+
+def thankyou():
+    global donors
+    print("\n------------------ THANK YOU ------------------\n")
+    print("- type 'list' to to see a complete list of donors -")
+    print("- type 'menu' at any time to return to the menu -\n")
+
+    # simple function to list all the donors
+    # calls the 'thankyou' function at the end to ask for user input
+    def list():
+        print("\n------------------ DONOR LIST ------------------\n")
+
+        for name in donors:
+            print(name[0])
+
+        thankyou()
+
+    # asks for a name from the user, or for the 'list' prompt
+    name_check = input("Enter first and last name of a donor\n\n-> ").lower()
+
+    if name_check == 'menu':
+        menu()
+    if name_check == 'list':
+        list()
+    else:
+        name_loc = 0
+
+        for name in donors:
+            if name_check == name[0].lower():
+                break
+            name_loc += 1
```

Comment:
you could use `for`/`else` instead of counter variable and append in the for loop once you find the existing donor